stable/kong/templates/migrations-post-upgrade.yaml

```diff
@@ -45,6 +46,17 @@ spec:
               key: postgresql-password
         command: [ "/bin/sh", "-c", "until nc -zv $KONG_PG_HOST $KONG_PG_PORT -w1; do echo 'waiting for db'; sleep 1; done" ]
       {{- end }}
+      {{- if .Values.cassandra.enabled }}
```

Comment:
Hm, we could get rid of the two initContainers, but since this chart can enable both the postgres and cassandra databases at the same time it makes things complicated.  To construct the correct command for each of the three options will probably require a helper template to be defined.  Can you think of an easier option to handle creating a bash command for the three options: postgres/cassandra/both?

To illustrate my point, this is a reasonably ugly way of doing what I'm describing:
```
{{- if or (.Values.postgresql.enabled) (.Values.cassandra.enabled)  }}
command: [ "/bin/sh", "-c", "until [[ nc -zv $KONG_PG_HOST $KONG_PG_PORT -w1 && $KONG_CASSANDRA_HOST $KONG_CASSANDRA_PORT -w1]]; do echo 'waiting for dbs'; sleep 1; done" ]
{{- end }}
{{- else .Values.postgresql.enabled }}
command: [ "/bin/sh", "-c", "until nc -zv $KONG_PG_HOST $KONG_PG_PORT -w1; do echo 'waiting for db'; sleep 1; done" ]
{{- end }}
{{- else .Values.cassandra.enabled }}
command: [ "/bin/sh", "-c", "until nc -zv $KONG_CASSANDRA_HOST $KONG_CASSANDRA_PORT -w1; do echo 'waiting for db'; sleep 1; done" ]
{{- end }}
```
I don't know the exact syntax for the first option, but it's something like that.

I think a helper template could output one of those three bash commands and then use that function in this template.  Let me know if that's the direction you think this should go, or if there's an easier alternative I'm not seeing.

Reply:
On further thought, perhaps the `nc` could only address the database that is set with the `env.database`.  Both databases may be enabled, but only the one that is being used will need to be bootstrapped.  This might simplify things a bit.  

It could all be done without a helper template, but because the port and host parameters are not simply `DB_HOST` and `DB_PORT`, there's still a bunch of if/else's to set a local variable based on which database is being used.  It's probably still clearer if it's taken care of in a helper template.

Reply:
While functionally it is possible to enable Postgres and Cassandra, enabling both the databases at the same time doesn't make any sense, so we can safely assume that they will never be enabled at the same time. We can use documentation to warn anyone to not do such a thing.

Reply:
Putting it in a helper template is in my todo list but haven't got around it, so if you do it, that would be great. Let's hae a small if block around postgres, cassandra for the value of DB_HOST and DB_PORT.

Also, I'll be off the grid for few weeks so please expect delayed responses. Thank you!

Reply:
@hbagdi Hope you had some pleasant off-the-grid time.

I simplified the initContainer block with some helper templates.  Let me know what you think.

I also removed the KONG_PG_PASSWORD variable entirely from the initContainer since netcat doesn't use it.

I could even remove the environment variables entirely from the initContainer section and simply define the entire command in a helper template.  However, I thought the way I checked in was simple enough and might actually be nice for people to see what was going on in the template.  I could go either way, just let me know.